### PR TITLE
osd: drop redundant assert in ECTransaction::generate_transactions

### PR DESCRIPTION
**Pull request by Songweibin**
Signed-off-by: songweibin <song.weibin@zte.com.cn>

### CODE REVIEW

File: src/osd/ECTransaction.cc [hunk 1]
```diff
@@ -196,9 +196,6 @@ void ECTransaction::generate_transactions(
       xattr_rollback[ECUtil::get_hinfo_key()] = old_hinfo;
       
       if (op.is_none() && op.truncate && op.truncate->first == 0) {
-	assert(op.truncate->first == 0);
-	assert(op.truncate->first ==
-	       op.truncate->second);
```
Comment (gregsfortytwo) [inline src/osd/ECTransaction.cc]: I’m not sure exactly what’s going on here, but we check for this equality below so I don’t think it’s redundant. (Maybe inaccurate?)

Did some other issue lead you here, or did you find it by code inspection? Did you look at the git history to see how it happened?

**Comment by Songweibin** [comment]
@gregsfortytwo  I was intending to drop the obviously redundant ```if (op.truncate->first != op.truncate->second)``` below...  
Fixed now! Would please help to have a look again? Thanks.

**Comment by xiexingguo** [comment]
QA run seems to be positive

**Comment by tchaikov** [comment]
@xiexingguo could you post the link to the qa run?

**Comment by xiexingguo** [comment]
> @xiexingguo could you post the link to the qa run?

Sure @tchaikov 

http://pulpito.ceph.com/xxg-2018-03-13_03:40:13-rados-wip-xxg-testing-distro-basic-smithi/
(The failure is not related, which is caused by one of my backfill-preempted fix)

**Comment by gregsfortytwo** [comment]
I don't suppose anybody did some git archeology to figure out where these redundant checks and asserts came from?
Getting rid of them may be great, but sometimes it's also indicative of a latent issue so it's always nice to understand what happened. :)

**Pull request closed without merging**